modules/photons_control/device_finder.py

```diff
@@ -704,11 +802,26 @@ async def matches(self, sender, fltr, collections):
 
         async def gen(reference, sender, **kwargs):
             for e in self.points_from_fltr(fltr):
+                if e == InfoPoints.VERSION:
+                    continue
+                if e.value.condition is not None and not e.value.condition(self):
+                    continue
                 if self.final_future.done():
                     return
                 if not self.point_futures[e].done():
                     yield e.value.msg
 
+        for e in self.points_from_fltr(fltr):
```

Comment:
?

Reply:
Attempt one billionth and 2.

Reply:
perhaps once we've sorted out the other comments I'll take a look at what's happening here. Can't remember how device_finder works off the top of my head well enough to remember what this does and why it does it.